### PR TITLE
Added condition to detect unit only if when mentioned in unit_type

### PR DESCRIPTION
**Pull request by amansrivastava17**

### CODE REVIEW

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by chiragjn** [comment]
This removes `Any` mode without any alternative for it. Is there any plan for it ?

**Comment by chiragjn** [comment]
Update tests and documentation

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 1]
```diff
@@ -227,6 +227,18 @@ def test_en_number_detection_for_decimal_number_with_scale_and_unit(self):
         number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
         number_dicts, original_texts = number_detector_object.detect_entity(message)
 
+        zipped = zip(number_dicts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: W1637 zip built-in referenced when not iterating

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 2]
```diff
@@ -148,12 +148,12 @@ def test_en_number_detection_for_integer_number(self):
         self.assertEqual(len(zipped), 1)
         self.assertIn(({'value': '100', 'unit': None}, u'100'), zipped)
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: Missing tests for`unit_type=None` or a different unit is given

For e.g.

`my monthly salary is 10.12k rupees`, `unit_type=None` must return [`10120`]
`my monthly salary is 10.12k rupees`, `unit_type='weight'` must return []

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge
